source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -474,12 +491,16 @@ Network::FilterStatus Filter::establishUpstreamConnection() {
     return Network::FilterStatus::StopIteration;
   }
 
-  const uint32_t max_connect_attempts = config_->maxConnectAttempts();
-  if (connect_attempts_ >= max_connect_attempts) {
-    getStreamInfo().setResponseFlag(StreamInfo::CoreResponseFlag::UpstreamRetryLimitExceeded);
-    cluster->trafficStats()->upstream_cx_connect_attempts_exceeded_.inc();
-    onInitFailure(UpstreamFailureReason::ConnectFailed);
-    return Network::FilterStatus::StopIteration;
+  if (!config_->backoffStrategy() &&
+      !Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.tcp_proxy_retry_on_different_event_loop")) {
```

Comment:
sry, I think the no-backoff config and setting false runtime guard test is missing.